### PR TITLE
Fix pthread launch

### PR DESCRIPTION
**Pull request by dianarg**
- Moves initialization of Profile (handshake with application) out of the constructor and calls it explicitly in geopm_pmpi_init
- Fix for test_profile_policy

### CODE REVIEW

File: test_integration/geopm_test_integration.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def test_report_and_trace_generation(self):
             trace = self._output.get_trace_data(node_name=nn)
             self.assertNotEqual(0, len(trace))
 
-    @unittest.skipUnless('mr-fusion' in socket.gethostname(), "This test only enabled on known working systems.")
+    @unittest.skipIf(socket.gethostname() in [], "This test is disabled enabled on some systems.")
```
Comment (bgeltz) [inline test_integration/geopm_test_integration.py]: ```in []``` seems wrong.

File: src/Profile.cpp [hunk 2]
```diff
@@ -572,4 +617,9 @@ namespace geopm
     {
         return m_tprof_table;
     }
+
+    void ProfileImp::enable_pmpi(void)
+    {
+
+    }
```
Comment (bgeltz) [inline src/Profile.cpp:624]: Is this supposed to be empty?

File: src/Profile.cpp [hunk 3]
```diff
@@ -110,21 +126,29 @@ namespace geopm
...
 #endif
-        std::string sample_key(key_base + "-sample");
-        std::string tprof_key(key_base + "-tprof");
+        std::string sample_key(m_key_base + "-sample");
+        std::string tprof_key(m_key_base + "-tprof");
         int shm_num_rank = 0;
 
-        init_prof_comm(std::move(comm), shm_num_rank);
+        init_prof_comm(std::move(m_comm), shm_num_rank);
+        std::string step;
```
Comment (bgeltz) [inline src/Profile.cpp:134]: Thanks for adding this.  It will help with future debug here.

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: src/Profile.cpp (hunk 2 above)
Comment (dianarg) [reply src/Profile.cpp:624]: yes, one reason why I don't love this workaround.  Only DefaultProfile implements any behavior

**Review by bgeltz — APPROVED** [review]

**Pull request merged**